scripts/git-hooks/pre-commit

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
```

Comment:
Fixing this file without changing `sh` to `bash` seems not possible, as we are using many non-standard features.

These are the errors that appear in we keep `sh`:

```^-- SC2112: 'function' keyword is non-standard. Delete it.


In scripts/git-hooks/pre-commit line 22:
    local exit_status=0
    ^---------------^ SC2039: In POSIX sh, 'local' is undefined.


In scripts/git-hooks/pre-commit line 38:
function go_linting() {
^-- SC2112: 'function' keyword is non-standard. Delete it.


In scripts/git-hooks/pre-commit line 39:
    dirs_to_analyze=()
                    ^-- SC2039: In POSIX sh, arrays are undefined.


In scripts/git-hooks/pre-commit line 42:
        dirs_to_analyze+=("$(dirname "${file}")" )
        ^-------------^ SC2039: In POSIX sh, += is undefined.
                         ^-----------------------^ SC2039: In POSIX sh, arrays are undefined.


In scripts/git-hooks/pre-commit line 44:
    mapfile -t dirs_to_analyze < <(printf '%s\n' "${dirs_to_analyze[@]}" | sort -u)
    ^-- SC2039: In POSIX sh, 'mapfile' is undefined.
                                 ^-- SC2039: In POSIX sh, process substitution is undefined.
                                                  ^-------------------^ SC2039: In POSIX sh, array references are undefined.


In scripts/git-hooks/pre-commit line 47:
        local exit_status=0
        ^---------------^ SC2039: In POSIX sh, 'local' is undefined.


In scripts/git-hooks/pre-commit line 48:
        for file in "${dirs_to_analyze[@]}"
                     ^-------------------^ SC2039: In POSIX sh, array references are undefined.


In scripts/git-hooks/pre-commit line 60:
function black_formatting()
^-- SC2112: 'function' keyword is non-standard. Delete it.
```